#167190398 Implement CRUD operations for articles

chuxmykel opened the pull request:
#### What does this PR do?
- This PR implements CRUD operations on the platform.

#### Description of Task to be completed?
- Create CRUD pages and implement the functionality
- Create a `404` page to give feedback to users on searching for non-existent articles

#### How should this be manually tested?
- visit the staging link that will be created from this PR and try the following CRUD operations
  - Create article `/create-article`
  - Read article `/articles/<slug>`
  - Update article `/articles/<slug>/edit`

#### Any background context you want to provide?
- Some work will still be done on the CSS. I need to sync with @cvjude to brainstorm on where to position the delete button.
- Work is done and E2E is in progress...
  - [Create article e2e video](https://drive.google.com/file/d/1iLCAQhS05KFqGqE0IOSx8LTpwO3CrTto/view?usp=sharing)
  - [Read article e2e video](https://drive.google.com/file/d/1LTyfoWoOdgJwCyOUndXLdC4m5BFGBYTL/view?usp=sharing)
  - [Update article e2e video](https://drive.google.com/file/d/1KuLrdYfTeMucoRzc-cKHQTVvxksNR-4o/view?usp=sharing)
  - [Delete article e2e video](https://drive.google.com/file/d/1oyfMuYgWtRPeQUdgvcf4WN0oEVBP1Lut/view?usp=sharing)
  - [404 e2e video](https://drive.google.com/file/d/1NapP7J4jwBljL0wNthAEwo6LAwtrnRZN/view?usp=sharing)

#### What are the relevant pivotal tracker stories?
- [#167190398](https://www.pivotaltracker.com/story/show/167190398)

#### Screenshots (if appropriate)

#### Questions:

chuxmykel committed:
feat(article CRUD): implement create article

- design UI
- post article to Endpoint

[Delivers #167190398]

chuxmykel committed:
feat(article CRUD): implement create article

- design UI
- post article to Endpoint

[Delivers #167190398]

chuxmykel committed:
feat(article CRUD): implement create article

- design UI
- post article to Endpoint

[Delivers #167190398]

chuxmykel committed:
feat(article CRUD): implement create article

- design UI
- post article to Endpoint

[Delivers #167190398]

chuxmykel committed:
feat(article CRUD): implement read and edit

- design UI
- fetch read and update endpoints

[Delivers #167190398]

chuxmykel committed:
feat(article CRUD): implement read and edit

- design UI
- fetch read and update endpoints

[Delivers #167190398]

chuxmykel committed:
feat(article CRUD): implement read and edit

- design UI
- fetch read and update endpoints

[Delivers #167190398]

chuxmykel committed:
feat(article CRUD): implement read and edit

- design UI
- fetch read and update endpoints

[Delivers #167190398]

chuxmykel committed:
feat(article CRUD): add 404 page

- design 404 page UI
- point to 404 page in case of wrong routes

[Delivers #167190398]

chuxmykel committed:
feat(article CRUD): add 404 page

- design 404 page UI
- point to 404 page in case of wrong routes

[Delivers #167190398]

chuxmykel committed:
feat(article CRUD): add 404 page

- design 404 page UI
- point to 404 page in case of wrong routes

[Delivers #167190398]

chuxmykel committed:
Merge branch 'develop' into ft/167190398-article-crud

chuxmykel committed:
feat(article CRUD): add 404 page

- design 404 page UI
- point to 404 page in case of wrong routes

[Delivers #167190398]

chuxmykel committed:
feat(article CRUD): add 404 page

- design 404 page UI
- point to 404 page in case of wrong routes

[Delivers #167190398]

chuxmykel committed:
feat(article CRUD): add 404 page

- design 404 page UI
- point to 404 page in case of wrong routes

[Delivers #167190398]

chuxmykel committed:
feat(article CRUD): implement delete functionality

- add delete button
- implement delete functionality

[Delivers #167190398]

chuxmykel committed:
feat(article CRUD): implement delete functionality

- add delete button
- implement delete functionality

[Delivers #167190398]

chuxmykel committed:
feat(article CRUD): implement delete functionality

- add delete button
- implement delete functionality

[Delivers #167190398]

chuxmykel committed:
feat(article CRUD): implement delete functionality

- add delete button
- implement delete functionality

[Delivers #167190398]

chuxmykel committed:
feat(article CRUD): finishing touches on css

- fix image display
- fix article width

[Delivers #167190398]

chuxmykel committed:
feat(article CRUD): finishing touches on css

- fix image display
- fix article width

[Delivers #167190398]

obayomi96 reviewed (commented):
Good job on pulling this off @chuxmykel , this is a job well done and I commend you on the use of editor.js framework as well. However, I noticed you did not add an end-to-end test for these features. Can you kindly do that @chuxmykel . 

Again great work.

encodedBicoding reviewed (approved):
LVGTM! Just fix the merge conflicts.

chuxmykel commented:
> Good job on pulling this off @chuxmykel , this is a job well done and I commend you on the use of editor.js framework as well. However, I noticed you did not add an end-to-end test for these features. Can you kindly do that @chuxmykel .
> 
> Again great work.

Sorry about that. I just got home and I'm currently on it. I'll add the link to the PR as soon as I'm done.

chuxmykel commented:
> LVGTM! Just fix the merge conflicts.

Thanks. I'm working on getting that done now.

chuxmykel committed:
feat(article CRUD): end to end testing

- write end to end tests for CRUD operations
- write end to end tests for 404 page

[Delivers #167190398]

chuxmykel committed:
feat(article CRUD): end to end testing

- write end to end tests for CRUD operations
- write end to end tests for 404 page

[Delivers #167190398]

chuxmykel committed:
Merge branch 'develop' into ft/167190398-article-crud

Lundii reviewed (approved):
LGTM

Lundii reviewed (commented):
Nice work with the task. I want to ask will it be possible to reduce the width of the featured image of the article so all images can display with a good proportion.

cvjude commented on src/views/Articles/ReadArticle/ReadArticle.js:
```diff
@@ -0,0 +1,201 @@
+              <div className="article-user center">
+                <div className="article-user-details">
+                  <div className="article-img-div center">
+                    <img
```
Nice work Chux, I think you should load a custom placeHolder image if no image data comes from the database.
Here is the one we've been using 'https://res.cloudinary.com/drdje1skj/image/upload/v1567427029/profile-placeholder_gvxkia.gif'

chuxmykel replied:
All right. I will do that now. Thanks

cvjude reviewed (approved):
LGTM

chuxmykel commented:
> Nice work with the task. I want to ask will it be possible to reduce the width of the featured image of the article so all images can display with a good proportion.

That is possible. But that will depend entirely on the PO

Lundii commented:
> > Nice work with the task. I want to ask will it be possible to reduce the width of the featured image of the article so all images can display with a good proportion.
> 
> That is possible. But that will depend entirely on the PO

Alright.

Hadeneekeh reviewed (approved):
LGTM!!!

chuxmykel committed:
feat(article CRUD): read article

- fix null image cases

[Delivers #167190398]

chiboycalix reviewed (approved):
Nice work chuks. This Looks good to me

obayomi96 reviewed (approved):
Looks good to me
